src/redis_rdb.cc

```diff
@@ -0,0 +1,450 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "redis_rdb.h"
+
+#include <string>
+#include <vector>
+
+#include "redis_string.h"
+#include "redis_list.h"
+#include "redis_set.h"
+#include "redis_hash.h"
+#include "redis_zset.h"
+#include "redis_bitmap.h"
+#include "db_util.h"
+
+namespace Redis {
+
+const int RDB_VERSION = 6;
+// always use the redis logical database 0
+const int SELECT_DB = 0;
+const int RDB_6BITLEN = 0;
+const int RDB_14BITLEN = 1;
+const int RDB_32BITLEN = 0x80;
+
+enum RDBOpCode: unsigned char {
+    RDB_OPCODE_EXPIRETIME = 253,  /* Expire time in seconds. */
```

Comment:
@ColinChamber Additionally you can have a look at this commit (https://github.com/apache/incubator-kvrocks/commit/1dcf225acad001175df0ea142b5e498f29d1b18b) and pay attention to `const` and `enum` naming to make relevant changes in your PR.